src/Servers/HttpSys/test/FunctionalTests/RequestHeaderTests.cs

```diff
@@ -114,4 +189,35 @@ private async Task SendRequestAsync(string address, string customHeader, string[
         await Task.Run(() => socket.Receive(response));
         socket.Dispose();
     }
+
+    private async Task<string> SendRequestAsync(string address, IHeaderDictionary headers)
+    {
+        var uri = new Uri(address);
+        StringBuilder builder = new StringBuilder();
+        builder.AppendLine("POST / HTTP/1.1");
+        builder.AppendLine("Connection: close");
+        builder.Append("HOST: ");
+        builder.AppendLine(uri.Authority);
+        foreach (var header in headers)
+        {
+            foreach (var value in header.Value)
+            {
+                builder.Append(header.Key);
+                builder.Append(": ");
+                builder.AppendLine(value);
+            }
+        }
+        builder.AppendLine();
+
+        byte[] request = Encoding.ASCII.GetBytes(builder.ToString());
+
+        using (Socket socket = new Socket(SocketType.Stream, ProtocolType.Tcp))
+        {
+            socket.Connect(uri.Host, uri.Port);
+            socket.Send(request);
+            byte[] response = new byte[1024 * 5];
+            await Task.Run(() => socket.Receive(response));
+            return Encoding.ASCII.GetString(response);
```

Comment:
Only the filled part of the buffer should be decoded, Receive returns how much was read.
```suggestion
            var read = await Task.Run(() => socket.Receive(response));
            return Encoding.ASCII.GetString(response, 0, read);
```